### ISSUE TITLE
`paginateSearchProductsAsAdmin` in Service Catalog sdk has new non-optional parameters

### ISSUE THREAD

**Issue by kcwinner** [aws/aws-sdk-js-v3#5397]
### Checkboxes for prior research

- [X] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest)
- [X] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js).
- [X] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js-v3/issues) and didn't find any solution.

### Describe the bug

We have been using `paginateSearchProductsAsAdmin` from `@aws-sdk/client-service-catalog` as below:
```
import { paginateSearchProductsAsAdmin } from "@aws-sdk/client-service-catalog";

...
...

  const searchProductsAsAdminPaginator = paginateSearchProductsAsAdmin(
    { client, pageSize: 20 },
    {
      Filters: { FullTextSearch: [productName] },
      PortfolioId: portfolioId,
    },
  );
```

As of `3.429.0` the `Filters` field introduced non-optional parameters (below)
```
export declare const ProductViewFilterBy: {
    readonly FullTextSearch: "FullTextSearch";
    readonly Owner: "Owner";
    readonly ProductType: "ProductType";
    readonly SourceProductId: "SourceProductId";
};
```

### SDK version number

@aws-sdk/client-service-catalog@3.429.0

### Which JavaScript Runtime is this issue in?

Node.js

### Details of the browser/Node.js/ReactNative version

v18.16.0

### Reproduction Steps

```
import { paginateSearchProductsAsAdmin, ServiceCatalogClient } from "@aws-sdk/client-service-catalog";

export async function getProductId(
  client: ServiceCatalogClient,
  portfolioId: string,
  productName: string,
): Promise<string> {
  const searchProductsAsAdminPaginator = paginateSearchProductsAsAdmin(
    { client, pageSize: 20 },
    {
      Filters: { FullTextSearch: [productName] },
      PortfolioId: portfolioId,
    },
  );

  for await (const page of searchProductsAsAdminPaginator) {
    if (page.ProductViewDetails) {
      for (const productViewDetail of page.ProductViewDetails) {
        if (productViewDetail.ProductViewSummary?.Name === productName) {
          return productViewDetail.ProductViewSummary.ProductId as string;
        }
      }
    }
  }

  throw new Error(`Didn't find product ${productName} in portfolio ${portfolioId}.`);
}

```

### Observed Behavior

```
error TS2739: Type '{ FullTextSearch: string[]; }' is missing the following properties from type 'Record<ProductViewFilterBy, string[]>': Owner, ProductType, SourceProductId

    at createTSError (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/index.ts:859:12)
    at reportTSError (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/index.ts:863:19)
    at getOutput (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/index.ts:1077:36)
    at Object.compile (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/index.ts:1433:41)
    at transformSource (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/esm.ts:400:37)
    at /home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/esm.ts:[27](https://github.com/Stedi/core/actions/runs/6590433874/job/17907013446#step:8:28)8:53
    at async addShortCircuitFlag (/home/runner/work/core/core/scripts/versions-cli/node_modules/ts-node/src/esm.ts:409:15)
    at async nextLoad (node:internal/modules/esm/loader:163:22)
    at async ESMLoader.load (node:internal/modules/esm/loader:603:20)
    at async ESMLoader.moduleProvider (node:internal/modules/esm/loader:457:11) {
  diagnosticCodes: [ 2739 ]
}
```

### Expected Behavior

I expected Owner, ProductType, and SourceProductId to remain optional parameters.

### Possible Solution

Make the fields optional. They don't appear to be required because data is returned as expected when using `@ts-ignore`

**Comment by yenfryherrerafeliz** [comment]
Hi @kcwinner, thanks for reporting this. This issue has been fixed in the latest release which by now is 3.433.0. So please update to this version and you should have not issues.

I am going to resolve this issue now but please feel free of opening a new one if is there anything else that we can assist you with.

Thanks!

**Comment by kcwinner** [comment]
Looks to be good! Ty. 

I did notice those properties seem to be gone now? I don't need them, just an observation
